app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/dataSetSection/DataValuePresenter.kt

```diff
@@ -0,0 +1,741 @@
+                            List<DataElementOperand>>>(
+                        Flowable.just<CategoryCombo>(categoryCombo),
+                        repository.getDataElements(categoryCombo, sectionName),
+                        repository.getCatOptions(sectionName, categoryCombo.uid()),
+                        repository.getDataValues(
+                            orgUnitUid,
+                            periodTypeName,
+                            periodId,
+                            attributeOptionCombo,
+                            sectionName
+                        ),
+                        repository.getGreyFields(sectionName),
+                        repository.compulsoryDataElements,
+                        Function6<CategoryCombo,
+                            List<DataElement>,
+                            Map<String, List<List<Pair<CategoryOption, Category>>>>,
+                            List<DataSetTableModel>,
+                            List<DataElementOperand>,
+                            List<DataElementOperand>,
+                            Sextet<CategoryCombo,
+                                List<DataElement>,
+                                Map<String, List<List<Pair<CategoryOption, Category>>>>,
+                                List<DataSetTableModel>,
+                                List<DataElementOperand>,
+                                List<DataElementOperand>>>
+                        { val0, val1, val2, val3, val4, val5 ->
+                            Sextet.create(
+                                val0,
+                                val1,
+                                val2,
+                                val3,
+                                val4,
+                                val5
+                            )
+                        }
+                    ).toObservable().blockingFirst()
+                }
+                .map { data ->
+                    var options: List<List<String>> = ArrayList()
+                    for ((_, value) in data.val2()) {
+                        options = getCatOptionCombos(value, 0, ArrayList(), null)
+                    }
+                    transformCategories = ArrayList()
+                    catOptionOrder = getCatOptionOrder(options)
+                    for ((_, value) in transformCategories(data.val2())) {
+                        transformCategories!!.addAll(value)
+                    }
+
+                    dataTableModel = DataTableModel.create(
+                        data.val1(), data.val3(), data.val4(),
+                        data.val5(), data.val0(), transformCategories
+                    )
+
+                    setTableData(dataTableModel!!)
+                }
+                .subscribeOn(schedulerProvider.io())
+                .observeOn(schedulerProvider.ui())
+                .subscribe(
+                    { quartet ->
+                        view.setTableData(
+                            quartet.val0(),
```

Comment:
in kotlin you can add a variable name so you could do something like:
variableName1 = quartet.val0(),
variableName2 = quartet.val1(),
...

Reply:
Also, in kotlin we can deconstruct an object
    ` val (varName1, varName2, varNmae3, varName4) = quartet`

and varName1 will have quartet.val0() value and so forth.